frontend/public/src/components/CourseProductDetailEnroll_test.js

```diff
@@ -266,18 +270,171 @@ describe("CourseProductDetailEnroll", () => {
+
+        const selectorBar = infobox.find(".more-dates-enrollment-list")
+        assert.isTrue(selectorBar.exists())
+      }
+    })
+  })
+  ;[[true], [false]].forEach(([flexPriceApproved]) => {
+    it(`shows the flexible pricing available link if the user does not have approved flexible pricing for the course run`, async () => {
+      courseRun["approved_flexible_price_exists"] = flexPriceApproved
+      courseRun["course"] = {
+        page: {
+          financial_assistance_form_url: "google.com"
+        }
+      }
+      isWithinEnrollmentPeriodStub.returns(true)
+      isFinancialAssistanceAvailableStub.returns(true)
+      const { inner } = await renderPage()
+
+      const enrollBtn = inner.find(".enroll-now").at(0)
+      assert.isTrue(enrollBtn.exists())
+      await enrollBtn.prop("onClick")()
+
+      const modal = inner.find(".upgrade-enrollment-modal")
+
+      const flexiblePricingLink = modal.find(".financial-assistance-link").at(0)
+      if (flexPriceApproved) {
+        assert.isFalse(flexiblePricingLink.exists())
+      } else {
+        assert.isTrue(flexiblePricingLink.exists())
+      }
+    })
+  })
+
+  it("CourseInfoBox renders the archived message if the course is archived", async () => {
+    const courseRun = {
+      ...makeCourseRunDetail(),
+      is_self_paced:    true,
+      enrollment_end:   null,
+      enrollment_start: moment()
+        .subtract(1, "years")
+        .toISOString(),
+      start_date: moment()
+        .subtract(10, "months")
+        .toISOString(),
+      end_date: moment()
+        .subtract(7, "months")
+        .toISOString(),
+      upgrade_deadline: null
+    }
+    const course = {
+      ...makeCourseDetailWithRuns(),
+      courseruns: [courseRun]
+    }
+
+    const entities = {
+      currentUser: currentUser,
+      enrollments: [],
+      courseRuns:  [courseRun],
+      courses:     [course]
+    }
+
+    const { inner } = await renderPage({
+      entities: entities
+    })
+
+    assert.isTrue(inner.exists())
+    const infobox = inner.find("CourseInfoBox").dive()
+    assert.isTrue(infobox.exists())
+
+    const archivedMessage = infobox.find("div.course-archived-message")
+    assert.isTrue(archivedMessage.exists())
+
+    const contentAvailabilityMessage = infobox.find(
+      "div.course-timing-message div.enrollment-info-text"
+    )
+    assert.isTrue(contentAvailabilityMessage.exists())
+    assert.isTrue(
+      contentAvailabilityMessage
+        .first()
+        .text()
+        .includes("Course content available anytime")
+    )
+  })
 
   it(`shows form based enrollment button when upgrade deadline has passed but course is within enrollment period`, async () => {
     isWithinEnrollmentPeriodStub.returns(true)
     courseRun.is_upgradable = false
-    const { inner } = await renderPage()
+    course.next_run_id = courseRun.id
 
-    sinon.assert.calledWith(
-      helper.handleRequestStub,
-      "/api/course_runs/?relevant_to=",
-      "GET"
```

Comment:
A bunch of these asserts are removed as this is only checking that the connect Request is working - this is making sure Redux knows to ask for this data which the shallow renderer can not see. I tried mocking this out with Mount, but it's so many layers down due to changes in the Redux-Query library and lack of changes to Enzyme. The long term solution here is to swap to Jest which uses full react rendering. Enzyme does not and as a result, you have to jump through hoops to get a full integration test.  here, I felt it was more than adequate to test the component using what is in the store.